src/main/kotlin/no/nav/familie/ef/personhendelse/inntekt/Inntektsendringer.kt

```diff
@@ -0,0 +1,54 @@
+package no.nav.familie.ef.personhendelse.inntekt
+
+import no.nav.familie.ef.personhendelse.client.OppgaveClient
+import no.nav.familie.ef.personhendelse.inntekt.vedtak.EfVedtakRepository
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.time.YearMonth
+
+@Component
+class Inntektsendringer(
+    val vedtakRepository: EfVedtakRepository,
+    val inntektClient: InntektClient,
+    val oppgaveClient: OppgaveClient
+) {
+
+    val secureLogger: Logger = LoggerFactory.getLogger("secureLogger")
+
+    fun beregnInntektsendringer() {
+        val personerMedVedtakList = vedtakRepository.hentAllePersonerMedVedtak()
+
+        secureLogger.info("Antall personer med aktive vedtak: ${personerMedVedtakList.size}")
+
+        for (ensligForsørgerVedtakhendelse in personerMedVedtakList) {
+            val response = inntektClient.hentInntektshistorikk(
+                ensligForsørgerVedtakhendelse.personIdent,
+                YearMonth.now().minusYears(1),
+                null
+            )
+            if (harEndretInntekt(response)) {
+                secureLogger.info("Person med behandlingId ${ensligForsørgerVedtakhendelse.personIdent} kan ha inntektsendringer. Skal opprette oppgave.")
+            }
+        }
+    }
+
+    fun harEndretInntekt(inntektshistorikkResponse: InntektshistorikkResponse): Boolean {
```

Comment:
Trenger vi å hente inntekten 1 år bak i tiden når vi kun sjekker siste 2 månedene sine inntekter? Eller er det sånt som kommer sen?

Reply:
Ja, jeg tror dette er noe som tar de fleste og viktigste tilfellene, da det ikke burde være altfor ofte noen endrer inntekten mer enn 2 måneder bak i tid. Jeg ser for meg å bruke versjonsnummeret til å se om det har kommet noen endringer på inntekt bak i tid.